raiden/tests/integration/test_matrix_transport.py

```diff
@@ -433,3 +426,57 @@ def test_join_invalid_discovery(
 
     transport.stop()
     transport.get()
+
+
+@pytest.mark.parametrize('matrix_server_count', [2])
+def test_matrix_cross_server(matrix_transports, retry_interval):
+    transport0, transport1 = matrix_transports
+
+    received_messages0 = set()
+    received_messages1 = set()
+
+    message_handler0 = MessageHandler(received_messages0)
+    message_handler1 = MessageHandler(received_messages1)
+    raiden_service0 = MockRaidenService(message_handler0)
+    raiden_service1 = MockRaidenService(message_handler1)
+
+    transport0.start(raiden_service0, message_handler0, '')
```

Comment:
Please, if transport is `start`ed on a test, also `stop` and possibly even `get` them (to re-raise any teardown exception) on end of test, to avoid errors like #3130